sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -288,9 +288,10 @@ case class TruncateTableCommand(
   override def run(sparkSession: SparkSession): Seq[Row] = {
     val catalog = sparkSession.sessionState.catalog
     if (!catalog.tableExists(tableName)) {
-      logError(s"table '$tableName' in TRUNCATE TABLE does not exist.")
+      throw new AnalysisException(s"Table '$tableName' in TRUNCATE TABLE does not exist.")
```

Comment:
@yhuai : Hive is doing the right thing of not throwing an exception when `IF EXISTS` keyword is used in `DROP TABLE` (irrespective of the table being there or not). If end user expects exception to be thrown, then one should not use the `IF EXISTS` in the command. Having `IF EXISTS` helps for writing pipelines which run daily and one wants "drop if exists or do nothing" semantics.